astropy/units/equivalencies.py

```diff
@@ -561,6 +564,268 @@ def mass_energy():
                         ], "mass_energy")
 
 
+@dataclass
```

Comment:
Just glancing at this quickly... Maybe you've thought about this and this is the better implementation, but this feels like a case for ``Enum``.

Reply:
Just looked at the docs for Enum, seems like a similar approach. Both Enum and Dataclasses are part of the standard library.